packages/recommend-react/src/__tests__/Recommend.test.tsx

```diff
@@ -0,0 +1,110 @@
+import { act, render, waitFor } from '@testing-library/react';
+import React from 'react';
+import '@testing-library/jest-dom';
+
+import { createMultiSearchResponse } from '../../../../test/utils';
+import {
+  createRecommendClient,
+  hit,
+} from '../../../../test/utils/createRecommendClient';
+import { Recommend, useRecommendContext } from '../Recommend';
+
+function createMockedRecommendClient() {
+  const recommendClient = createRecommendClient({
+    getRecommendations: jest.fn(() =>
+      Promise.resolve(
+        createMultiSearchResponse({
+          hits: [hit, hit],
+        })
+      )
+    ),
+  });
+
+  return {
+    recommendClient,
+  };
+}
+
+describe('Recommend', () => {
```

Comment:
What I would expect from this test is:

- render only provided components (without Recommend client) and assert that the correct Recommend client method is called
- render provided components (without Recommend client) and detached components (with Recommend client) within the provider and assert that the correct Recommend client methods are called on the correct clients.

Basically, the tests should explain the contract as they written in the RFC.